### DISCUSSION TITLE
streams mode using the REST API

### DISCUSSION THREAD

**Discussion by dattnertravelzap** [redpanda-data/connect#1013] [Q&A]
When I run streams mode I'm using resources to bring in environment variables using this:

HugsETL.exe -r ./resources/staticfhir_sandbox.yaml streams  ./stream_config/webhooks.yaml

If I want to load the stream config using a post to the REST API to localhost:4195/streams 

how the heck do I have the config pull in my resource file with environment variables? Is this even possible?

thanks, David.

**Comment by Jeffail** [comment]
hey @dattnertravelzap, there's an API endpoint for creating individual resources: https://www.benthos.dev/docs/guides/streams_mode/streams_api#post-resourcestypeid, and the config provided there would replace environment variables for you. However, there's no endpoint right now for posting a resource config file potentially containing multiple resources, if you need that feel free to open an issue for it as it's something we can add.

**Comment by dattnertravelzap** [comment]
Great thanks! I'm a huge fan of Benthos! Keep up the great work and I'm
really enjoying the YouTube videos!

I'll let you know if we get stuck on multiple resources through the API.

Also, it would be cool if you could post an env file through the API that
would be used by one more stream configs so that as a pipeline is running
you could switch out environments on the fly and like point to dev for
testing from prod lets say while you're troubleshooting. I guess you could
do that with resources but my manager likes the environment file approach
better.

Thanks, David.



On Fri, Dec 17, 2021 at 2:37 AM Ashley Jeffs ***@***.***>
wrote:

> hey @dattnertravelzap <https://github.com/dattnertravelzap>, there's an
> API endpoint for creating individual resources:
> https://www.benthos.dev/docs/guides/streams_mode/streams_api#post-resourcestypeid,
> and the config provided there would replace environment variables for you.
> However, there's no endpoint right now for posting a resource config file
> potentially containing multiple resources, if you need that feel free to
> open an issue for it as it's something we can add.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Jeffail/benthos/discussions/1013#discussioncomment-1833883>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWZ3AG3XBVJLGUKNJ3SF6W3URMHFVANCNFSM5KHXZHQA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
<https://travelzap.com/>
*David Dattner | .net Architect*
travelzap | 29 E. Madison | Chicago, IL

*e: ***@***.*** ***@***.***> |* w: *travelzap.com
<https://travelzap.com/>
*call: *(773) 312- 5376 <+17733125376>
